flink-runtime/src/main/java/org/apache/flink/runtime/execution/ExecutionState.java

```diff
@@ -70,6 +70,8 @@ public enum ExecutionState {
 
     RECONCILING,
 
+    PENDING,
```

Comment:
How about handle `pending-operators` in the `JobDetailsHandler#createJobDetailsInfo`. This way, we wouldn't need to introduce any meaningless statuses to a common enum class.

Reply:
Agree with you. Since it’s not a valid status for stream jobs, how about including it in the serialized JSON stream graph (`JsonPlanGenerator#generateJsonStreamGraph`)? This would help avoid adding interfaces like `getPendingOperatorCounts()` to the `ExecutionGraph` and `ArchivedExecutionGraph` for processing in `JobDetailsHandler#createJobDetailsInfo`. And it will be processed in frontend code.

Reply:
I prefer to put the `pending-operators` field in the `JobDetailsInfo` because it seems more related to job execution concepts and cannot be fully represented by the stream graph plan.

Reply:
Okay, I have added `pending-operator-counts` to `JobDetailsInfo`. Since `JobDetailsHandler` needs to handle `pendingOperatorCounts`, I have also added a `getPendingOperatorCounts()` function to `AccessExecutionGraph`.